### DISCUSSION TITLE
How to access $<prop> with auto import

### DISCUSSION THREAD

**Discussion by volivieri-2-OD** [nuxt/nuxt#16258] [Questions]
Hello, I generate my API using openapi and I instantiate it with the correct configuration using the following code: 

```
  const runtimeConfig = useRuntimeConfig();

  const api = new DefaultApi(
    new Configuration({
      basePath: runtimeConfig.public.API_BASE_URL,
    }),
  );
```

I would like to be able to use `$api` directly in my code without any kind of import. What's the best solution? 

I tried 2 things:
- I created a plugin that provide $api. However, I have to write `const { $api } = useNuxtApp()` in each file.
- I created an $api composable. However, this creates a new instance each time the composable is called and the composable is not a variable.

I could mix the two solutions by having a plugin that provide $api and a composable that return the provided $api, but it seems a bit verbose and means I'll have to call $api() instead of accessing it.

Thanks

**Comment by TheAlexLichter** [comment]
You can use a composable to keep global state too. Something like:

```ts
let api = null

export const useApi = () => {
	api = // setup here

	return { api }
}
